src/states/elder/mod.rs

```diff
@@ -102,6 +102,8 @@ pub struct Elder {
     pfx_is_successfully_polled: bool,
     /// Peers we will disconnect from in the future.
     delayed_disconnects: HashMap<u64, PublicId>,
+    /// DKG cache
+    dkg_cache: BTreeMap<BTreeSet<PublicId>, EldersInfo>,
```

Comment:
I don't know, it means that we have to have 2 special cases, while here we push that info in the cache as needed and get it from it when ready. Seem simpler?

> Also, shouldn't this be in SharedState?
I don't think so. I don't think we should have a DKG running when we create a new Parsec. (This might be something we need to be careful about in Prune, but it would not seem sensible to initiate pruning if we are already having a DKG on to rest Parsec).

Reply:
> Seem simpler?

I uses more memory and It breaks the [Single Source of Truth](https://en.wikipedia.org/wiki/Single_source_of_truth) rule, but maybe it's something we can live with for now?